bundle/src/main/java/com/adobe/acs/commons/packagegarbagecollector/PackageGarbageCollectionJob.java

```diff
@@ -0,0 +1,216 @@
+import java.time.LocalDateTime;
+import java.time.Period;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+import static com.adobe.acs.commons.packagegarbagecollector.PackageGarbageCollectionScheduler.GROUP_NAME;
+import static com.adobe.acs.commons.packagegarbagecollector.PackageGarbageCollectionScheduler.MAX_AGE_IN_DAYS;
+
+@Component(
+    service = JobConsumer.class,
+    immediate = true,
+    property = { JobConsumer.PROPERTY_TOPICS + "=" + PackageGarbageCollectionScheduler.JOB_TOPIC })
+public class PackageGarbageCollectionJob implements JobConsumer {
+    public static final DateTimeFormatter LOCALIZED_DATE_FORMATTER = DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM);
+    private static final Logger LOG = LoggerFactory.getLogger(PackageGarbageCollectionJob.class);
+
+    private static final String SERVICE_USER = "package-garbage-collection";
+
+    @Reference
+    Packaging packaging;
+
+    @Reference
+    ResourceResolverFactory resourceResolverFactory;
+
+    @Override
+    public JobResult process(Job job) {
+        Session session;
+        String groupName = job.getProperty(GROUP_NAME, String.class);
+        Integer maxAgeInDays = job.getProperty(MAX_AGE_IN_DAYS, Integer.class);
+        int packagesRemoved = 0;
+        LOG.debug("Job Configuration: ["
+                + "Group Name: {}, "
+                + "Service User: {}, "
+                + "Age of Package {} days,]", groupName, SERVICE_USER, maxAgeInDays);
+
+        try (ResourceResolver resourceResolver = resourceResolverFactory.getServiceResourceResolver(
+            Collections.singletonMap(ResourceResolverFactory.SUBSERVICE, SERVICE_USER))) {
+            session = resourceResolver.adaptTo(Session.class);
+            JcrPackageManager packageManager = packaging.getPackageManager(session);
+            List<JcrPackage> packages = packageManager.listPackages(groupName, false);
+
+            for (JcrPackage jcrPackage : packages) {
+                String packageDescription = getPackageDescription(jcrPackage);
+                if (isPackageOldEnough(jcrPackage, maxAgeInDays)) {
+                    if (!isLatestInstalled(jcrPackage, packages)) {
+                        packageManager.remove(jcrPackage);
+                        packagesRemoved++;
+                        LOG.info("Deleted package {}", packageDescription);
+                    } else {
+                        LOG.info("Not removing package because it's the current installed one {}", packageDescription);
+                    }
+                } else {
+                    LOG.debug("Not removing package because it's not old enough {}", packageDescription);
+                }
+            }
+        } catch (LoginException | RepositoryException | IOException e) {
+            LOG.error("Unable to clear packages", e);
```

Comment:
You might have cleared packages already when you reach this point. It could make sense to log this information as well, when you fail the job here.
Also, is it possible to log the information at which package the process failed?

Reply:
Will log an info level message to show when a package is started to be checked, and if any packages have been removed - I'll log this in this catch block.